### DISCUSSION TITLE
Support to Fluent-UI react controls (office-ui-fabric-react)

### DISCUSSION THREAD

**Discussion by joaojmendes** [react-hook-form/react-hook-form#2244] [Q&A]
Please advise how to use with Fluent-UI React Controls, give some examples.

Thank you!

**Comment by bluebill1049** [comment]
provide a codesandbox see if we can help.

**Reply by PathToSharePoint** [reply]
If it helps, the following page has a detailed sample and there's a button to export it to CodeSandbox
https://fluentsite.z22.web.core.windows.net/0.53.0/components/form/definition

Note that there are multiple flavors of Fluent UI. The above one is called Fluent UI Northstar.

**Reply by PathToSharePoint** [reply]
btw in case you're not familiar with it, Fluent UI Northstar is the UI framework provided by Microsoft for Microsoft Teams customizations. It's relatively new but with significant potential for growth.

I am not familiar with React Hook Form myself but of course I've heard of it, and I was looking for an example to see if it could help with my Teams forms.

**Reply by PathToSharePoint** [reply]
A fresh article on the topic:
https://spblog.net/post/2021/04/13/how-to-use-react-hook-form-together-with-fluent-ui-react-aka-office-ui-fabric-react

**Reply by TharakaSl** [reply]
@PathToSharePoint Did you find any example for Fluent UI NorthStar with react-hook-form ?

**Reply by PathToSharePoint** [reply]
I didn't. I ended up building a simpler form with minimal validation and not relying on a form library.

**Reply by TharakaSl** [reply]
Thanks. Found a sample from Fluent repository.
https://github.com/microsoft/fluentui/blob/master/packages/fluentui/react-northstar-prototypes/src/prototypes/FormValidation/FormHooks.tsx

**Reply by PathToSharePoint** [reply]
Nice! Thanks for sharing.
